client/service/src/task_manager/tests.rs

```diff
@@ -315,92 +319,90 @@ fn ensure_task_manager_future_continues_when_childs_not_essential_task_fails() {
 }
 
 struct TestLayer {
-	spans_entered: Arc<Mutex<Vec<String>>>,
-	spans: Arc<Mutex<std::collections::HashMap<Id, String>>>,
+	spans_found: Arc<Mutex<Option<Vec<Id>>>>,
 }
 
-impl<S: Subscriber> Layer<S> for TestLayer {
-	fn new_span(&self, attrs: &Attributes<'_>, id: &Id, _ctx: Context<S>) {
-		self.spans.lock().insert(id.clone(), attrs.metadata().name().to_string());
-	}
-
-	fn on_record(&self, _: &Id, _: &Record<'_>, _: Context<S>) {}
+impl<S> Layer<S> for TestLayer
+where
+	S: Subscriber + for<'a> LookupSpan<'a>,
+{
+	fn on_event(&self, _: &Event<'_>, ctx: Context<S>) {
+		let mut spans_found = self.spans_found.lock();
 
-	fn on_event(&self, _: &Event<'_>, _: Context<S>) {}
+		if spans_found.is_some() {
+			panic!("on_event called multiple times");
+		}
 
-	fn on_enter(&self, span: &Id, _: Context<S>) {
-		let name = self.spans.lock().get(span).unwrap().clone();
-		self.spans_entered.lock().push(name);
+		*spans_found = Some(ctx.scope().map(|x| x.id()).collect());
 	}
-
-	fn on_exit(&self, _: &Id, _: Context<S>) {}
-
-	fn on_close(&self, _: Id, _: Context<S>) {}
 }
 
-type TestSubscriber = tracing_subscriber::layer::Layered<
-	TestLayer,
-	tracing_subscriber::fmt::Subscriber
->;
-
 fn setup_subscriber() -> (
-	TestSubscriber,
-	Arc<Mutex<Vec<String>>>,
+	impl Subscriber + for<'a> LookupSpan<'a>,
+	Arc<Mutex<Option<Vec<Id>>>>,
 ) {
-	let spans_entered = Arc::new(Mutex::new(Default::default()));
+	let spans_found = Arc::new(Mutex::new(Default::default()));
 	let layer = TestLayer {
-		spans: Arc::new(Mutex::new(Default::default())),
-		spans_entered: spans_entered.clone(),
+		spans_found: spans_found.clone(),
 	};
 	let subscriber = tracing_subscriber::fmt().finish().with(layer);
-	(subscriber, spans_entered)
+	(subscriber, spans_found)
 }
 
+/// This is not an actual test, it is used by the `telemetry_span_is_forwarded_to_task` test.
+/// The given test will call the test executable and only execute this one test that
+/// test that the telemetry span and the prefix span are forwarded correctly. This needs to be done
+/// in a separate process to avoid interfering with the other tests.
 #[test]
-fn telemetry_span_is_forwarded_to_task() {
-	let (subscriber, spans_entered) = setup_subscriber();
-	let _sub_guard = tracing::subscriber::set_global_default(subscriber);
+fn log_something() {
```

Comment:
Please give this a more meaningful name that makes it easier to see that this is connected to the other test.

Reply:
Done in 
95e4742